src/util/mdb-exportjson.c

```diff
@@ -0,0 +1,186 @@
+	while (1) {
+		if (is_binary) {
+			if (!bin_len--)
+				break;
+		} else /* use \0 sentry */
+			if (!*value)
+				break;
+
+		if (quote_len && !strncmp(value, quote_char, quote_len)) {
+			fprintf(outfile, "%s%s", escape_char, quote_char);
+			value += quote_len;
+#ifndef DONT_ESCAPE_ESCAPE
+		} else if (orig_escape_len && !strncmp(value, escape_char, orig_escape_len)) {
+			fprintf(outfile, "%s%s", escape_char, escape_char);
+			value += orig_escape_len;
+#endif
+		} else if (*value < 0x20) {
+			if (!is_binary || drop_nonascii) {
+				putc(' ', outfile);
+				++value;
+			} else {
+				// escape control codes / binary data.
+				fprintf(outfile, "\\x%02x", *(unsigned char*)value++);
+			}
+		} else {
+			putc(*value++, outfile);
+		}
+	}
+	fputs(quote_char, outfile);
+}
+
+static void
+print_col(FILE *outfile, char* col_name, gchar *col_val, int col_type, int bin_len) {
+	print_quoted_value(outfile, col_name, -1);
+	fputs(separator_char, outfile);
+	if (is_quote_type(col_type)) {
+		if (!is_binary_type(col_type)) {
+			bin_len = -1;
+		}
+		print_quoted_value(outfile, col_val, bin_len);
+	} else
+		fputs(col_val, outfile);
+}
+int
+main(int argc, char **argv)
+{
+	unsigned int i;
+	MdbHandle *mdb;
+	MdbTableDef *table;
+	MdbColumn *col;
+	char **bound_values;
+	int  *bound_lens;
+	FILE *outfile = stdout;
+	drop_nonascii = 1;
+	int  opt;
+	char *value;
+	size_t length;
+
+	while ((opt=getopt(argc, argv, "AD:"))!=-1) {
+		switch (opt) {
+		case 'A':
+			drop_nonascii = 0;
+		break;
+		case 'D':
+			mdb_set_date_fmt(optarg);
+		break;
+		default:
+		break;
+		}
+	}
+	/*
+	** optind is now the position of the first non-option arg,
+	** see getopt(3)
+	*/
+	if (argc-optind < 2) {
+		fprintf(stderr,"Usage: %s [options] <file> <table>\n",argv[0]);
+		fprintf(stderr,"where options are:\n");
+		fprintf(stderr,"  -D <format>          set the date format (see strftime(3) for details)\n");
+		fprintf(stderr,"  -A                   drop non ascii characters in non-binary fields\n");
+		exit(1);
+	}
+
+	if (!(mdb = mdb_open(argv[optind], MDB_NOFLAGS))) {
+		exit(1);
+	}
+
+	table = mdb_read_table_by_name(mdb, argv[argc-1], MDB_TABLE);
+	if (!table) {
+		fprintf(stderr, "Error: Table %s does not exist in this database.\n", argv[argc-1]);
+		mdb_close(mdb);
+		exit(1);
+	}
+
+	/* read table */
+	mdb_read_columns(table);
+	mdb_rewind_table(table);
+
+	bound_values = (char **) g_malloc(table->num_cols * sizeof(char *));
+	bound_lens = (int *) g_malloc(table->num_cols * sizeof(int));
+	for (i=0;i<table->num_cols;i++) {
+		/* bind columns */
+		bound_values[i] = (char *) g_malloc0(MDB_BIND_SIZE);
+		mdb_bind_column(table, i+1, bound_values[i], &bound_lens[i]);
+	}
+
+	while(mdb_fetch_row(table)) {
+		fputs(row_start, outfile);
+		int add_delimiter = 0;
+		for (i=0;i<table->num_cols;i++) {
+			if (add_delimiter)
+				fputs(delimiter, outfile);
+			add_delimiter = 0;
+			col=g_ptr_array_index(table->columns,i);
+			if (bound_lens[i]) {
```

Comment:
`if (add_delimiter)` block should be inside `if (bound_lens[i]) {`, otherwise unnecessary comma may be in output, e.g.

```
{"key":"value",}
```